src/App.tsx

```diff
@@ -1,61 +1,124 @@
 import './App.scss';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Post } from './types/post';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getNewPostId(posts: Post[]) {
+  const maxId = Math.max(...posts.map(post => post.id));
+
+  return maxId + 1;
+}
+
+export const App: React.FC = () => {
+  const [posts, setPosts] = useState(todos);
+
+  function addTodosInArr({ id, ...data }: Post) {
+    const newPost = {
+      id: id || getNewPostId(posts),
+      ...data,
+    };
+
+    setPosts(currentPosts => [...currentPosts, newPost]);
+  }
```

Comment:
[CODE STYLE] - The function name `addTodosInArr` is not very descriptive and could be improved. Consider renaming it to something more descriptive like `addPostToPosts`.